python_modules/dagster/dagster/_utils/error.py

```diff
@@ -239,3 +239,84 @@ def serializable_error_info_from_exc_info(
     else:
         tb_exc = traceback.TracebackException(exc_type, e, tb)
         return _serializable_error_info_from_tb(tb_exc)
+
+
+DAGSTER_FRAMEWORK_SUBSTRINGS = [
+    "/site-packages/dagster/",
+    "/python_modules/dagster/dagster/",
+]
+
+IMPORT_MACHINERY_SUBSTRINGS = [
+    "importlib/__init__.py",
+    "importlib._bootstrap",
+]
+
+
+def unwrap_user_code_error(error_info: SerializableErrorInfo) -> SerializableErrorInfo:
+    """Extracts the underlying error from the passed error, if it is a DagsterUserCodeLoadError."""
+    if error_info.cls_name == "DagsterUserCodeLoadError":
+        return unwrap_user_code_error(error_info.cause)
+    return error_info
+
+
+def remove_non_user_code_lines_from_serializable_exc_info(
+    error_info: SerializableErrorInfo,
+):
+    return remove_matching_lines_from_serializable_exc_info(
+        error_info,
+        DAGSTER_FRAMEWORK_SUBSTRINGS + IMPORT_MACHINERY_SUBSTRINGS,
+    )
+
+
+def remove_matching_lines_from_serializable_exc_info(
+    error_info: SerializableErrorInfo,
```

Comment:
`serializable_exc_info` slight mismatch of exc_info -> ErrorInfo 
also since its in the type hint i think you dont need to spell the whole thing out